components/Chat.js

```diff
@@ -1,26 +1,89 @@
-import { StyleSheet, View, Text } from "react-native";
-import { useEffect } from "react";
+import {
+  StyleSheet,
+  View,
+  Text,
+  KeyboardAvoidingView,
+  Platform,
+} from "react-native";
+import { useEffect, useState } from "react";
+import { GiftedChat, Bubble } from "react-native-gifted-chat";
 
 const Chat = ({ route, navigation }) => {
   // these variables are getting passed from Start screen
   const { name, color } = route.params;
 
+  const [messages, setMessages] = useState([]);
+
+  // append the new messages to the previous message list
+  const onSend = (newMessages) => {
+    setMessages((previousMessages) =>
+      GiftedChat.append(previousMessages, newMessages)
+    );
+  };
+
+  // creates custom chat bubble
+  const renderBubble = (props) => {
+    return (
+      <Bubble
+        {...props}
+        wrapperStyle={{
+          right: {
+            backgroundColor: "#000",
+          },
+          left: {
+            backgroundColor: "#FFF",
+          },
+        }}
+      />
+    );
+  };
+
+  useEffect(() => {
+    setMessages([
+      {
+        _id: 1,
+        text: "Hello developer",
+        createdAt: new Date(),
+        user: {
+          _id: 2,
+          name: "React Native",
+          avatar: "https://i.pravatar.cc/300",
+        },
+      },
+      {
+        // sets the system message
+        _id: 2,
+        text: "Welcome to the chat",
+        createdAt: new Date(),
+        system: true,
+      },
+    ]);
+  }, []);
+
   useEffect(() => {
     navigation.setOptions({ title: name });
   }, []);
 
   return (
-    <View style={[styles.container, { backgroundColor: color }]}>
-      <Text>Hello Chat screen</Text>
+    <View style={styles.container}>
```

Comment:
The color of the Chat screen no longer changes. You need to use `color` extracted from the route params.